### PR TITLE
Fix autoload section from composer.json and allow to use multiple paths

### PR DESCRIPTION
**Pull request by sidz**
Fix for https://github.com/infection/infection/issues/23 issue.

I've added some rules + few exceptions.
Didn't find any valid way to parse composer.json without including JsonLint library. I think it'll be overhead.

### CODE REVIEW

File: tests/Config/Guesser/SourceDirGuesserTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Infection\Tests\Config\Guesser;
+
+use Infection\Config\Guesser\SourceDirGuesser;
+use PHPUnit\Framework\TestCase;
+
+class SourceDirGuesserTest extends TestCase
```
Comment (borNfreee) [inline tests/Config/Guesser/SourceDirGuesserTest.php]: Just found another `SourceDirGuesserTest` file at the incorrect path (`src/tests/Guesser`).

Could you please move tests from that file to your file and remove it?

**Comment by borNfreee** [comment]
> Didn't find any valid way to parse composer.json without including JsonLint library. I think it'll be overhead.

You are right, I would avoid including a new dependency as well because of problems outlined in #21

PR looks good to me, thank you! Will be merged today if nobody has any other comments.

File: tests/Config/Guesser/SourceDirGuesserTest.php (hunk 1 above)
Comment (sidz) [reply tests/Config/Guesser/SourceDirGuesserTest.php]: @borNfreee done

**Comment by borNfreee** [comment]
Thank you @sidz !

**Pull request merged**